oil_lang/expr_eval.py

```diff
@@ -441,6 +460,41 @@ def _ToInteger(self, val):
         raise ValueError("%r doesn't look like an integer" % val)
 
     raise ValueError("%r isn't like an integer" % (val,))
+
+  def _ValueToInteger(self, val):
+    # type: (value_t) -> int
+    UP_val = val
+    with tagswitch(val) as case:
+      if case(value_e.Int):
+        val = cast(value__Int, UP_val)
+        return val.i
+
+      elif case(value_e.Str):
+        val = cast(value__Str, UP_val)
+        if match.LooksLikeInteger(val.s):
+          return int(val.s)
+        else:
+          raise ValueError("%r doesn't look like an integer" % val.s)
+
+    raise error.InvalidType('Expected Int', loc.Missing())
+
+  def _MaybeConvertStr(self, val):
```

Comment:
I think the names were ToInteger and ToNumber (float, int)?  How about preserving that, like `_ValueToNumber()` ?

Reply:
ah sure. i can send a follow up for this